controller/resource.go

```diff
@@ -81,3 +82,34 @@ func (c *ResourceController) Register(ctx *app.RegisterResourceContext) error {
 
 	return ctx.Created(&app.RegisterResourceResponse{ResourceID: &res.ResourceID})
 }
+
+// Scopes runs the scopes action, which returns the available scopes for the specified resource
+func (c *ResourceController) Scopes(ctx *app.ScopesResourceContext) error {
+	// Load the identity from the context
+	currentIdentity, err := login.LoadContextIdentityIfNotDeprovisioned(ctx, c.app)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+
+	// Check if the resource exists
+	err = c.app.ResourceRepository().CheckExists(ctx, ctx.ResourceID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+
+	// Load the scopes
+	pc, err := c.app.PrivilegeCacheService().CachedPrivileges(ctx, currentIdentity.ID, ctx.ResourceID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+
+	res := &app.ResourceScopesData{
+		Data: &app.ResourceScopes{
+			ID:     ctx.ResourceID,
+			Type:   "resource",
```

Comment:
Should be of type "scopes", what do you think @xcoulon ?

Reply:
right, it makes sense to use `scopes` here, indeed.

Reply:
@sbose78 @xcoulon This doesn't make sense according to the [JSON API](http://jsonapi.org/format/#document-resource-object-identification) specification:

> Within a given API, each resource object’s type and id pair MUST identify a single, unique resource. 

We are not identifying a scope here, we are identifying a resource.  Technically, we could make the case that the scopes are an *attribute* of the resource, however I think this is a special case as this endpoint produces a dynamic result depending on who is doing the calling, which is why I don't think that using attributes fits here.

Reply:
ah, I missed that in the design: the endpoint response is something like 
````
{
    "data": {
        "id": "74372086-9b16-4cef-a100-70e3e2a32194",
        "scopes": ["foo", "bar"],
        "type": "resource"
    }
}
````
but I'm not sure that having a `scopes` attribute as a child of `data` is valid either, as explained in here:http://jsonapi.org/format/#document-resource-objects. My understanding is that the `id` and `type` elements are mandatory, and `attributes`, `relationships`, `links` and `meta` are optional, but no other element is allowed. So I would suggest moving the `scopes` under the `attributes` element. As for the `type` value to use, maybe `resource_scopes` would be more accurate? My reasoning is that the `ID` refers to the resource, response does not contain the resource's attributes, but rather an array of scopes (based on the user's roles, indeed).

Reply:
My reluctance to put `scopes` under the `attributes` element is because it is not a static attribute of the resource, rather a *calculated* set of values dependent on the current user, and as such not really an attribute of the resource itself.  

As for the `type` value, `resource_scopes` is still not compliant with the JSON API spec - the identifier is for a resource, and the spec does not allow us to stray from this.  I.e. the "MUST" in:

> **MUST** identify a single, unique resource.

is final and doesn't allow for any other alternative.  Just because we don't include other *usual* attributes of the resource in the request doesn't mean it's still not resource-centric.

Back on where to place the `scopes` value - if we were worried about strict compliance here, then I suggest we place the `scopes` under the [meta](http://jsonapi.org/format/#document-meta) tag instead.

Reply:
The API `GET /api/resource/<resource_id>/scopes` is returning scopes for a specific resource - which is why I feel `scopes` could be more appropriate or as @xcoulon mention resource_scopes could be a good option.